src/test/scala/scorex/core/DefaultModifiersCacheSpecification.scala

```diff
@@ -0,0 +1,120 @@
+package scorex.core
+
+import org.scalatest.{Matchers, PropSpec}
+import org.scalatest.prop.{GeneratorDrivenPropertyChecks, PropertyChecks}
+import scorex.core.consensus.{History, HistoryReader, ModifierSemanticValidity, SyncInfo}
+import scorex.core.consensus.History.ModifierIds
+import scorex.core.serialization.Serializer
+import scorex.crypto.hash.Blake2b256
+
+import scala.util.Try
+
+class DefaultModifiersCacheSpecification extends PropSpec
```

Comment:
Would like to see more tests here (and may be in a ScorexTestkit):
- Application of the same modifier twice
- Remove of modifier, that is not in a cache
- Addition of incorrect modifier id to `rememberedKeys`
- Application of modifier, with a key in `rememberedKeys`